internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -717,16 +707,31 @@ func (r *AgentServiceConfigReconciler) newAssistedServiceDeployment(log logrus.F
 		},
 	}
 
-	// User is responsible for knowing to restart assisted-service
-	annotations := instance.ObjectMeta.GetAnnotations()
-	configmapName, ok := annotations[configmapAnnotation]
+	// User is responsible for:
+	// - knowing to restart assisted-service when specifying the configmap via annotation
+	// - removing the annotation when the configmap is deleted
+	userConfigName, ok := instance.ObjectMeta.GetAnnotations()[configmapAnnotation]
 	if ok {
-		log.Infof("ConfigMap %s from namespace %s being used to configure assisted-service deployment", configmapName, r.Namespace)
+		log.Infof("ConfigMap %s from namespace %s being used to configure assisted-service deployment", userConfigName, r.Namespace)
+		// even though we aren't modifying the assisted-service, we want to recognize
+		// when the applied config differs from the running config
+		userConfigHash, err = r.getCMHash(ctx, types.NamespacedName{Name: userConfigName, Namespace: r.Namespace})
+		if err != nil {
+			// We will take no explicit action when the user provided configMap
+			// is not found. But we will log a warning and emit a 'Warning' event
+			if !errors.IsNotFound(err) {
+				return nil, nil, err
+			}
+			msg := fmt.Sprintf("A ConfigMap was specified on the agent via annotation '%s', but that ConfigMap was not found", configmapAnnotation)
+			r.Recorder.Event(instance, "Warning", "UserConfigMapNotFound", msg)
```

Comment:
@mhrivnak I remember the last time I made something an event it wasn't appropriate. This feels like an appropriate time to emit one. What do you think?

Reply:
Was originally planning to cover all of this configmap stuff in a single PR. Decided against it.

Reply:
Not seeing this in the diff, so maybe you changed your mind? :) Let me know if there's something I should look at.